optional_plugins/gtester/avocado_gtester/__init__.py

```diff
@@ -0,0 +1,149 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2018
+# Authors: Amador Pahim <apahim@redhat.com>
+
+"""
+Plugin to run GLib Test Framework tests in Avocado
+"""
+
+import re
+
+from avocado.utils import path as utils_path
+from avocado.utils import process
+
+from avocado.core import exceptions
+from avocado.core import loader
+from avocado.core import output
+from avocado.core import test
+from avocado.core.plugin_interfaces import CLI
+
+
+try:
+    _GTESTER_BIN = utils_path.find_command('gtester')
+except utils_path.CmdNotFoundError:
+    _GTESTER_BIN = None
+
+
+class GtesterTest(test.SimpleTest):
+
+    """
+    Run a Gtester command as a SIMPLE test.
+    """
+
+    def __init__(self, name, params=None, base_logdir=None, job=None):
+        super(GtesterTest, self).__init__(name, params, base_logdir, job)
+
+    @property
+    def filename(self):
+        """
+        Returns the path of the gtester test suite.
+        """
+        return self.name.name.split(':')[0]
+
+    def test(self):
+        """
+        Create the Gtester command and execute it.
+        """
+
+        if _GTESTER_BIN is None:
+            raise exceptions.TestError("gtester binary not found")
+
+        test_name = self.name.name.split(':')[1]
+        cmd = '%s -p=%s %s' % (_GTESTER_BIN, test_name, self.filename)
+        result = process.run(cmd, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail('Gtester Test execution returned a '
+                      'non-0 exit code (%s)' % result)
+
+
+class NotGtesterTest(object):
+
+    """
+    Not a gtester test (for reporting purposes)
+    """
+
+
+class GtesterLoader(loader.TestLoader):
+    """
+    Gtester loader class
+    """
+    name = "gtester"
+
+    def __init__(self, args, extra_params):
+        super(GtesterLoader, self).__init__(args, extra_params)
+
+    def discover(self, url, which_tests=loader.DEFAULT):
+        if _GTESTER_BIN is None:
+            raise exceptions.TestError("gtester binary not found")
+
+        avocado_suite = []
+        subtests_filter = None
+
+        if url is None:
+            return []
+
+        if ':' in url:
+            url, _subtests_filter = url.split(':', 1)
+            subtests_filter = re.compile(_subtests_filter)
+        try:
+            cmd = '%s --quiet -l %s' % (_GTESTER_BIN, url)
+            result = process.run(cmd)
+        except Exception as details:
+            if which_tests == loader.ALL:
+                return [(NotGtesterTest, {"name": "%s: %s" % (url, details)})]
+            return []
+
+        for test in result.stdout.splitlines():
+            test_name = "%s:%s" % (url, test)
+            try:
+                # Avoiding duplicates
```

Comment:
Can you please create a card to attempt to fix the issue upstream and reference it here?

Reply:
ok